src/Internal/DependencyManagement/Definition.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * An extension to the Definition class to prevent constructor injection from being possible.
+ *
+ * @package Automattic\WooCommerce\Internal\DependencyManagement
+ */
+
+namespace Automattic\WooCommerce\Internal\DependencyManagement;
+
+use \League\Container\Definition\Definition as BaseDefinition;
+
+/**
+ * An extension of the definition class that replaces constructor injection with method injection.
+ */
+class Definition extends BaseDefinition {
```

Comment:
It seemed useful that we entirely disable constructor injection to prevent mistakes. We can use this class in place of thephpleague/container's `Definition` class to accomplish that task.

Reply:
Sure, but then we need to actually disallow constructors with arguments, see my comment in `ExtendedContainer.php`.